business/src/main/java/org/mskcc/cbio/portal/service/ApiService.java

```diff
@@ -546,26 +528,31 @@ else if(type.equals("frequency"))
 	}
 
 	@Transactional
+  @PreAuthorize("hasPermission(#study_id, 'CancerStudy', 'read')")
 	public List<DBSample> getSamples(String study_id) {
 		return sampleMapperLegacy.getSamplesByStudy(study_id);
 	}
 
 	@Transactional
+  @PreAuthorize("hasPermission(#study_id, 'CancerStudy', 'read')")
 	public List<DBSample> getSamplesBySample(String study_id, List<String> sample_ids) {
 		return sampleMapperLegacy.getSamplesBySample(study_id, sample_ids);
 	}
 
 	@Transactional
+  @PreAuthorize("hasPermission(#study_id, 'CancerStudy', 'read')")
 	public List<DBSample> getSamplesByPatient(String study_id, List<String> patient_ids) {
 		return sampleMapperLegacy.getSamplesByPatient(study_id, patient_ids);
 	}
 
 	@Transactional
+  @PostFilter("hasPermission(filterObject.id, 'CancerStudy', 'read')")
 	public List<DBStudy> getStudies() {
 		return studyMapperLegacy.getAllStudies();
 	}
 
 	@Transactional
+  @PreAuthorize("hasPermission(#study_ids, 'List<CancerStudyId>', 'read')")
```

Comment:
See my response to your comment in CancerStudyPermissionEvaluator.  It results in 403 error.  Not an empty list.  We are assuming that it is malicious in this case.